editor/plugins/script_text_editor.cpp

```diff
@@ -2419,6 +2420,8 @@ void ScriptTextEditor::register_editor() {
 	ED_SHORTCUT("script_text_editor/contextual_help", TTR("Contextual Help"), KeyModifierMask::ALT | Key::F1);
 	ED_SHORTCUT_OVERRIDE("script_text_editor/contextual_help", "macos", KeyModifierMask::ALT | KeyModifierMask::SHIFT | Key::SPACE);
 
+	ED_SHORTCUT("script_text_editor/lookup_symbol", TTR("Lookup Symbol"), Key::F12);
```

Comment:
This needs a sensible override for macOS, since F-buttons are not always easily accessible on that platform. Alternatively, we could add this shortcut without binding any key combination to it.

Reply:
That's a valid point, however, IMO user can define the key they like，so this might not be very important I guess?
And also, my mac showing that visual studio also use F12 for both mac and windows. And mac user can always switch to fn-off-by-default mode.
So I think make it costomizable to keep persistentency with other shortcut might be better?

Reply:
It's already customizable with your PR. However, we provide alternative bindings for macOS to avoid conflicts and work around macOS UX. Namely, F-keys are problematic for touchbar devices. You can see this is done in the same file and all around the codebase, and it needs to be done here as well.